src/coreclr/src/jit/morph.cpp

```diff
@@ -6395,6 +6395,14 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
                 tree->SetOper(GT_IND);
                 tree->AsOp()->gtOp1 = addr;
 
+                bool plsSpeculative = true;
+                if ((info.compCompHnd->getStaticFieldCurrentClass(symHnd, &plsSpeculative) != NO_CLASS_HANDLE) &&
+                    !plsSpeculative)
+                {
+                    JITDUMP("Marking static read-only field '%s' as invariant.\n", eeGetFieldName(symHnd));
+                    tree->gtFlags |= GTF_IND_INVARIANT;
+                }
+
```

Comment:
This will only kick in for 64 bit targets. Also, did you look at how often we end up in the "else" clause below on a 64 bit target? 

Seems like we could handle those cases as well, by introducing `GTF_CLS_VAR_INVARIANT` and making associated changes elsewhere.

Reply:
> Also, did you look at how often we end up in the "else" clause below on a 64 bit target?
I've just tested it on a pretty heavy app (with IL-only version of CoreLib) and it was:
if block - 9568 calls
else block - 43 calls

Do you think it worth optimizing them as well?